pkg/types/interfacetests/codec_interface_tests.go

```diff
@@ -0,0 +1,377 @@
+package interfacetests
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+	ocrTypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+type EncodeRequest struct {
+	TestStructs  []TestStruct
+	ExtraField   bool
+	MissingField bool
+	TestOn       string
+}
+
+type ChainReaderInterfaceTester interface {
+	Setup(t *testing.T)
+	Teardown(t *testing.T)
```

Comment:
I think this could be simplified away, since `Setup` has the opportunity to use `t.Cleanup` already.
```suggestion
```
It should enable simpler implementations too, since a local closure can be used instead of storing a field to be used from another method.

Reply:
codec and tests have been ripped out of this pr